### PR TITLE
Refactor outbound request write logic

### PR DESCRIPTION
**Pull request by shafreenAnfar**

### CODE REVIEW

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/AbstractHTTPAction.java [hunk 1]
```diff
@@ -229,34 +220,40 @@ private void send(Context context, HTTPCarbonMessage outboundRequestMsg,
...
     /**
      * Serlaize multipart entity body. If an array of body parts exist, encode body parts else serialize body content
      * if it exist as a byte channel.
      *
-     * @param context                   Represent the ballerina context which is the runtime state of the program
-     * @param outboundRequestMsg        Outbound request that needs to be sent across the wire
-     * @param httpClientConnectorLister Represent http client connector listener
-     * @param boundaryString            Boundary string that should be used in encoding body parts
+     * @param context         Represent the ballerina context which is the runtime state of the program
+     * @param boundaryString  Boundary string that should be used in encoding body parts
```
Comment (chamil321) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/AbstractHTTPAction.java]: Shall we document "messageOutputStream" here?

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/AbstractHTTPAction.java [hunk 2]
```diff
@@ -270,28 +267,21 @@ private BStruct getEntityStruct(Context context) {
     /**
      * Encode body parts with the given boundary and send it across the wire.
      *
-     * @param outboundRequestMsg        Outbound request message
-     * @param httpClientConnectorLister Represent http client connector listener
-     * @param boundaryString            Boundary string of multipart entity
-     * @param entityStruct              Represent ballerina entity struct
+     * @param boundaryString Boundary string of multipart entity
```
Comment (chamil321) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/AbstractHTTPAction.java]: Shall we document "messageOutputStream" here as well

File: pom.xml [hunk 3]
```diff
@@ -1183,7 +1183,7 @@
         <equinox.osgi.version>3.10.2.v20150203-1939</equinox.osgi.version>
         <equinox.osgi.services.version>3.4.0.v20140312-2051</equinox.osgi.services.version>
         <carbon.kernel.version>5.1.0</carbon.kernel.version>
-        <transport.http.version>6.0.94</transport.http.version>
+        <transport.http.version>6.0.97</transport.http.version>
```
Comment (riyafa) [inline pom.xml]: We can have it as 0.95 otherwise it shall fail because of compilation failure because of keep-alive change.

Comment (shafreenAnfar) [reply pom.xml]: 0.95 has a build break..

**Pull request merged**